core/domain/user_services.py

```diff
@@ -1651,11 +1638,15 @@ def get_usernames(user_ids, strict=False):
             non_system_user_ids.append(user_id)
 
     non_system_users_settings = get_users_settings(
-        non_system_user_ids, strict=strict)
+        non_system_user_ids, strict=strict, include_marked_deleted=True)
 
     for index, user_settings in enumerate(non_system_users_settings):
         if user_settings:
-            usernames[non_system_user_indices[index]] = user_settings.username
+            usernames[non_system_user_indices[index]] = (
+                user_settings.username
+                if not user_settings.deleted
+                else USER_IDENTIFICATION_FOR_USER_BEING_DELETED
```

Comment:
USER_IDENTIFICATION --> USERNAME? (seems different from user ID)

Though I'm kind of worried about including this at all, I think someone is going to treat this as a regular username one day and break things. Could you instead return a 2-tuple perhaps? That said, why do we need to include the "users who are being deleted" at all in this function?

In either case please make sure the Returns docstring is very clearly updated to explain what's being returned.

Reply:
I mean this is similar to when we return the pseudonymous username in this function, with the introduction of wipeout we cannot really guarantee that this function will always return usernames that can "do" all the stuff, also we sometimes use "admin" or "OppiaMigrationBot" username which also cannot "do" that much stuff as a regular user with username.